brainframe_qt/api_utils/streaming/stream_manager.py

```diff
@@ -1,139 +1,144 @@
 import logging
+import typing
+from typing import Dict, Optional
+
+from PyQt5.QtCore import QObject, QThread
 
-from brainframe.api import BrainFrameAPI, StatusReceiver
 from brainframe.api.bf_codecs import StreamConfiguration
 from gstly import gobject_init
 from gstly.gst_stream_reader import GstStreamReader
 from gstly.abstract_stream_reader import StreamReader
 
+from brainframe_qt.api_utils import api
+
 from .synced_reader import SyncedStreamReader
 
 
-class StreamManager:
-    """Keeps track of existing Stream objects, and creates new ones as
-    necessary.
-    """
+class StreamManager(QObject):
+    """Keeps track of existing Stream objects, and creates new ones as necessary"""
+
+    REHOSTED_VIDEO_TYPES = [
+        StreamConfiguration.ConnType.WEBCAM,
+        StreamConfiguration.ConnType.FILE,
+    ]
+    """Video types that are re-hosted by the server"""
+
+    def __init__(self, *, parent: QObject):
+        super().__init__(parent=parent)
+
+        self._stream_readers: Dict[int, SyncedStreamReader] = {}
 
-    REHOSTED_VIDEO_TYPES = [StreamConfiguration.ConnType.WEBCAM,
-                            StreamConfiguration.ConnType.FILE]
-    """These video types are re-hosted by the server."""
+        self._init_signals()
 
-    def __init__(self, status_receiver: StatusReceiver):
-        self._stream_readers = {}
-        self._status_receiver = status_receiver
-        self._async_closing_streams = []
-        """A list of StreamReader objects that are closing or may have finished
-        closing"""
+    def _init_signals(self) -> None:
+        self.destroyed.connect(self.close)
```

Comment:
I assume that `destroyed` is only called when the window closes?

Reply:
`destroyed` is called when Qt's garbage collector begins destroying the QObject

Reply:
Sure, but is that only expected when the whole system shuts down, or is there another case in the client where it would get destroyed?